lms/djangoapps/courseware/masquerade.py

```diff
@@ -71,27 +71,36 @@ def handle_ajax(request, course_key_string):
     group_id = request_json.get('group_id', None)
     user_partition_id = request_json.get('user_partition_id', None) if group_id is not None else None
     user_name = request_json.get('user_name', None)
+    # The 'user_name' param is actually a string representing either the username or the user's
+    # email address. 'found_user_name' represents the actual username property for a User.
+    found_user_name = None
+    # This view should return an error if no 'user_name' was provided, but many different test cases
+    # rely on the view still working if 'user_name' is blank or None.
     if user_name:
         users_in_course = CourseEnrollment.objects.users_enrolled_in(course_key)
```

Comment:
I'd like to propose a better method of finding the user, using Django's `Q()` query feature. This is more succinct than both of the old and the new queries.

`users_in_course.get(email=user_name).username` -> `users_in_course.get(Q(uername=user_name) | Q(email=user_name)).username`